src/sentry/event_manager.py

```diff
@@ -1880,6 +1880,101 @@ def _calculate_span_grouping(jobs, projects):
             sentry_sdk.capture_exception()
 
 
+@metrics.wraps("save_event.get_or_create_performance_group")
+def _get_or_create_performance_group(jobs, projects):
+    # TODO: batch operations (like rate limiting) so we don't repeat for each job
+    for job in jobs:
+        # FEATURE FLAG RANDOMIZER
+        # check options to see if we should be creating performance issues for this org
+        rate = options.get("incidents-performance.rollout-rate")
+        if (rate and rate > random.random()) or True:
+
+            # TODO: get fingerprint
+            existing_grouphash = job.get("fingerprint")
+            project = job["event"].project
+
+            # use fingerprint to check if group exists
+            if existing_grouphash is None:
+
+                # GROUP DOES NOT EXIST
+                with sentry_sdk.start_span(
+                    op="event_manager.create_group_transaction"
+                ) as span, metrics.timer(
+                    "event_manager.create_group_transaction"
+                ) as metric_tags, transaction.atomic():
+                    span.set_tag("create_group_transaction.outcome", "no_group")
+                    metric_tags["create_group_transaction.outcome"] = "no_group"
+
+                    try:
+                        short_id = project.next_short_id()
+                    except OperationalError:
+                        metrics.incr(
+                            "next_short_id.timeout",
+                            tags={"platform": "unknown"},
+                        )
+                        sentry_sdk.capture_message("short_id.timeout")
+                        raise HashDiscarded("Timeout when getting next_short_id")
+
+                    # TODO: RATE LIMITER
+                    # from sentry.sentry_metrics.configuration import UseCaseKey
+                    # from sentry.sentry_metrics.indexer.base import KeyCollection
+                    # from sentry.sentry_metrics.indexer.ratelimiters import writes_limiter
+                    # with writes_limiter.check_write_limits(UseCaseKey("performance"), KeyCollection({job.organization.id: {job.organization.name}})) as writes_limiter_state:
+                    #     pass
+
+                    group = Group.objects.create(
+                        project=project,
+                        short_id=short_id,
+                        first_release_id=None,
+                    )
```

Comment:
It might be tricky, but if possible it'd be good to try and refactor this logic so that it's shared with error issue creation as much as possible. Things we need to make sure happen here:
 - Create rows in `GroupRelease` and `GroupEnvironment`
 - We should set `firstRelease` on the group if one exists
 - When creating an error issue, we pass a bunch of **kwargs to fill in extra fields. I don't know what data is in there, but likely we need at least some of it

Reply:
ah ok. yeah I just cut most of that out because I didn't understand it haha. I had the `kwargs` in there as well but removed them. will get on it for this